File: Infrastructure-Module/persistence/src/main/java/com/bamyanggang/persistence/strongpoint/mapper/StrongPointMapper.java
package com.bamyanggang.persistence.strongpoint.mapper;

import com.bamyanggang.domainmodule.domain.strongpoint.aggregate.StrongPoint;
import com.bamyanggang.persistence.strongpoint.jpa.entity.StrongPointJpaEntity;
import org.springframework.stereotype.Component;

@Component
public class StrongPointMapper {
    public StrongPointJpaEntity toJpaEntity(StrongPoint strongPoint) {
        return new StrongPointJpaEntity(
                strongPoint.getId(),
                strongPoint.getName(),
                strongPoint.getUserId()
        );
    }

    public StrongPoint toDomainEntity(StrongPointJpaEntity strongPointJpaEntity) {
        return new StrongPoint(
                strongPointJpaEntity.getStrongPointId(),
                strongPointJpaEntity.getName(),
                strongPointJpaEntity.getUserId()
        );
    }
}

File: Infrastructure-Module/persistence/src/main/java/com/bamyanggang/persistence/experience/mapper/StrongPointMapper.kt
package com.bamyanggang.persistence.experience.mapper

import com.bamyanggang.domainmodule.domain.experience.aggregate.StrongPoint
import com.bamyanggang.persistence.experience.jpa.entity.StrongPointJpaEntity
import org.springframework.stereotype.Component

@Component
class StrongPointMapper {
    fun toJpaEntity(strongPoint: StrongPoint) =
        StrongPointJpaEntity.of(
            strongPoint.id,
            strongPoint.name,
            strongPoint.userId
        )

    fun toDomain(strongPointJpaEntity: StrongPointJpaEntity) =
        StrongPoint.toDomain(
            strongPointJpaEntity.id,
            strongPointJpaEntity.name,
            strongPointJpaEntity.userId
        )
}